### PR TITLE
Adds support for annotation parsing

### PR DESCRIPTION
**Pull request by artman**
- Annotations are supported on: Typdefs, Consts, Enums, Enum values, Structs (& Unions, Exceptions), Struct Fields, Services and Functions
- Updates test to stringify and subsequently parse the Thrift doc in order to get rid of undefined values
- Adds extensive unit test for annotations
- Adds correct error messages when annotations aren't syntactically correct

This PR opts to diverge from the coding style of the rest of the library regarding annotations. Annotations are declared as optional parameters and default to undefined instead of null. This has the following upsides:

- Undefined values aren't omitted when stringifying to JSON
- Existing unit tests continue to pass

### CODE REVIEW

**Comment by kevin-greene-ck** [comment]
@artman It's looking good. Thanks for the effort.

File: src/main/parser.ts [hunk 1]
```diff
@@ -389,6 +392,32 @@ export function createParser(tokens: Array<Token>, report: ErrorReporter = noopR
...
+  // Annotations → '(' (Identifier '=' StringConstant ListSeparator?)* ')'
+  function parseAnnotations(): Array<Annotation> | null {
+    if (check(SyntaxType.LeftParenToken)) {
+      const annotations: Array<Annotation> = []
+      advance()
+      while (!check(SyntaxType.RightParenToken)) {
+          const nameToken: Token = consume(SyntaxType.Identifier)
+          consume(SyntaxType.EqualToken)
+          const next: Token = advance()
+          const value: StringConstant = parseStringValue(next)
```
Comment (kevin-greene-ck) [inline src/main/parser.ts]: I am curious what we gain from adding StringConstant instead of using a StringLiteral as the value of an Annotation.

Comment (artman) [reply src/main/parser.ts]: You're right, there's no need to add a new StringConstant type. Wasn't very familiar with the code-base, so thought we'd want to return constant types like Floats and Doubles do, but given that e.g. include definitions just use StringLiterals, I'll remove the StringConstant.

**Review by kevin-greene-ck — APPROVED** [review]

**Comment by kevin-greene-ck** [comment]
Excellent work, I'll get this out in a release today.

**Comment by artman** [comment]
Great, thanks!

**Pull request merged**